src/Core/Types/Utilities/XmlDocumentationExtensions.cs

```diff
@@ -0,0 +1,514 @@
+        }
+
+        /// <summary>Returns the contents of the "summary" XML documentation tag for the specified member.</summary>
+        /// <param name="member">The reflected member.</param>
+        /// <returns>The contents of the "summary" tag for the member.</returns>
+        internal static async Task<string> GetXmlSummaryAsync(this MemberInfo member)
+        {
+            return await GetXmlDocumentationTagAsync(member, "summary").ConfigureAwait(false);
+        }
+
+        /// <summary>Returns the contents of the "returns" or "param" XML documentation tag for the specified parameter.</summary>
+        /// <param name="parameter">The reflected parameter or return info.</param>
+        /// <returns>The contents of the "returns" or "param" tag.</returns>
+        internal static async Task<string> GetXmlDocumentationAsync(this ParameterInfo parameter)
+        {
+            var assemblyName = parameter.Member.Module.Assembly.GetName();
+            using (Lock.Lock())
+            {
+                if (IgnoreAssembly(assemblyName))
+                    return string.Empty;
+
+                var documentationPath = GetXmlDocumentationPath(parameter.Member.Module.Assembly);
+                var element = await GetXmlDocumentationWithoutLockAsync(parameter, documentationPath).ConfigureAwait(false);
+                return RemoveLineBreakWhiteSpaces(GetXmlDocumentationText(element));
+            }
+        }
+
+        /// <summary>Clears the cache.</summary>
+        /// <returns>The task.</returns>
+        internal static Task ClearCacheAsync()
+        {
+            using (Lock.Lock())
+            {
+                Cache.Clear();
+                return Task.CompletedTask;
+            }
+        }
+
+        /// <summary>Returns the contents of an XML documentation tag for the specified member.</summary>
+        /// <param name="type">The type.</param>
+        /// <param name="tagName">Name of the tag.</param>
+        /// <returns>The contents of the "summary" tag for the member.</returns>
+        private static Task<string> GetXmlDocumentationTagAsync(this Type type, string tagName)
+        {
+            return GetXmlDocumentationTagAsync((MemberInfo)type.GetTypeInfo(), tagName);
+        }
+
+        /// <summary>Converts the given XML documentation <see cref="XElement"/> to text.</summary>
+        /// <param name="element">The XML element.</param>
+        /// <returns>The text</returns>
+        private static string GetXmlDocumentationText(this XElement element)
+        {
+            if (element == null)
+            {
+                return null;
+            }
+
+            var value = new StringBuilder();
+            foreach (var node in element.Nodes())
+            {
+                var currentElement = node as XElement;
+                if (currentElement == null)
+                {
+                    value.Append(node);
+                    continue;
```

Comment:
I updated this function to use `continue` statements because it had a rather deep nesting structure to accomplish this work. If you're opposed to using `continue` statements and structures, I can revert that change. I found this to be more readable.

Original source: https://github.com/RicoSuter/NJsonSchema/blob/master/src/NJsonSchema/Infrastructure/XmlDocumentationExtensions.cs#L138-L181

Reply:
I think this Is better than the original!